src/utils/bindtwostepcarettoattribute.js

```diff
@@ -380,6 +397,8 @@ class TwoStepCaretHandler {
 			if ( position.isAtStart ) {
 				if ( this._hasSelectionAttribute ) {
 					this._removeSelectionAttribute();
```

Comment:
Why aren't we preventing caret move in this case? 🤔

Reply:
Would that matter in this case? It's a start of the block anyway and we're moving left.

Reply:
Doesn't it matter in case of `<paragraph>foo</paragraph><paragraph><$text a="1">{}foo</$text></paragraph>`? Won't we get to the previous paragraph if we won't stop caret movement?

Reply:
I tested this in manual test and for some reason it works 🤷‍♂️

Reply:
Hm, you're right. It should be prevented. But the funny thing is that it works. As long as the `keydown` event is stopped.

Reply:
lol :)

![apr-18-2018 16-01-40](https://user-images.githubusercontent.com/1502228/38936233-a41d451a-4320-11e8-8eb1-3db3fd5f8eef.gif)